src/contracts/middleware/IndexRegistry.sol

```diff
@@ -211,12 +215,9 @@ contract IndexRegistry is IIndexRegistry {
         if (totalOperatorsHistoryLength == 0) {
             return 0;
         }
-                
-        // if `blockNumber` is in future, return current number of operators
-        if (blockNumber > uint32(block.number)) {
-            return _totalOperatorsHistory[quorumNumber][totalOperatorsHistoryLength - 1].index;
-        // else if `blockNumber` is from before the `quorumNumber` existed, return `0`
-        } else if (blockNumber < _totalOperatorsHistory[quorumNumber][0].fromBlockNumber) {
+
+        // if `blockNumber` is from before the `quorumNumber` existed, return `0`
+        if (blockNumber < _totalOperatorsHistory[quorumNumber][0].fromBlockNumber) {
             return 0;
         }
```

Comment:
am I crazy or is this function missing the "meat" that actually calculates the correct value?

Reply:
I was crazy, deleting lmao